### ISSUE TITLE
why ExecutionQueue not designed as a polling mechanism?

### ISSUE THREAD

**Issue by DengJin88** [apache/incubator-brpc#1164]
Hi, we know that ExecutionQueue works as a batching mechanism,say we sumbit some tasks and the "execute" runs in one bthread,after processing all the tasks in "iter",it exits the bthread and waits for a next batch tasks,and runs again(and it allows to run the tasks "in place",while it still has a "start-process-exit" lift cycle).

Nowdays,polling is popular in area such as intergrating with SPDK,I'd like to write code like following,making the "execute" always looping in one bthread(maybe this idea conflict with bthread design target:it won't do a long runing task while not allow a re-scheduling?):

```
int execute(void *meta, bthread::IteratorTask<std::string> &iter) {
  while (iter.is_queue_stopped() == false) {
    for (;iter; ++iter) {
      // submit IO task with spdk
    }
   // submition finished,and I'd like to polling spdk result
   // spdk polling
  }
  return 0;
}
```

while in current ExecutionQueue implemetation,I have to start a another bthread doing polling and send fake task to the ExecutionQueue to trigger SPDK polling thing,is this the right way to use ExectionQueue?

**Comment by DengJin88** [comment]
ping @jamesge @llhe

**Comment by jamesge** [comment]
ExecutionQueue is designed for running tasks (with dependencies) **in order**, and different ExecutionQueues run in parallel to make good use of CPU cores.  The polling version does not make sense because we're not subscribing "events" that are not controlled by us: the submitted tasks are just created by users, who can perfectly run the tasks one-by-one with a for-loop, without any other code. However, if one of the tasks blocks, the calling thread blocks. If you don't need this behavior, you need ExecutionQueue.
